lua/mason-nvim-dap/mappings/adapters/init.lua

```diff
@@ -0,0 +1,52 @@
+--- Lazily map debug adapters
+
+local M = {}
+
+---Convenience function for creating a set from a list.
+---@param list table A list which's values are to be used as indexes in the set.
+---@return table A table with the values from the list set as indexes.
+local function Set(list)
+	local set = {}
+	for _, val in ipairs(list) do
+		set[val] = true
+	end
+	return set
+end
+
+---Set of adapters that can be mapped directly from this module.
+---I.e. They should be able to be found with for example:
+--->lua
+---	require('mason-nvim-dap.mappings.adapters.chrome')
+---<
+---This should be updated if files with adapter mappings are added or removed
+---from 'lua/mason-nvim-dap/mappings/adapters/'.
+local OWN_MAPS = Set({
+	'bash',
+	'chrome',
+	'codelldb',
+	'coreclr',
+	'cppdbg',
+	'dart',
+	'delve',
+	'firefox',
+	'kotlin',
+	'mix_task',
+	'node2',
+	'php',
+	'python',
+})
+
+local META = {}
+function META.__index(table, key)
+	if not OWN_MAPS[key] then
+		return nil
+	end
+
+	local adapter = require('mason-nvim-dap.mappings.adapters.' .. key)
+	table[key] = adapter
```

Comment:
can't we just use `pcall`?
```suggestion
	local adapter, ok = pcall(require, 'mason-nvim-dap.mappings.adapters.' .. key)
	if !ok then
	    return nil
	end
	table[key] = adapter
```

Reply:
But why? It's a lot slower and adds a security risk by adding somewhere for people to inject code. Both of those are probably completely neglible but what is the benefit? Less lines of code? Specifying which modules are adapters also allows for potentially adding submodules which aren't in the future.

Reply:
> It's a lot slower and adds a security risk by adding somewhere for people to inject code

the risk remains with or without this. a person can just inject code into an existing file and have it run. 

Less maintenance overhead. A single place to update when adding a new adapter. I looked at the performance and seemed negligible about 10ms diff

Reply:
re-run a few times. actually performance diff is like 2ms

Reply:
I did say that while worse both the performance and security were likely completely negligible but so is probably the amount saved on maintenance.

I did consider using pcall first but went with this solution since I felt it was faster and safer. But I think it's more important that you go with the solution that you prefer since it is your repository. Worst case it can simply be changed later.